tck/src/main/java/org/eclipse/microprofile/fault/tolerance/tck/telemetryMetrics/util/TelemetryHistogramMetric.java

```diff
@@ -32,31 +42,126 @@
  * Most tests should use {@link TelemetryMetricGetter} to create instances of this class.
  */
 public class TelemetryHistogramMetric {
+
+    public static final List<Double> EXPECTED_BOUNDARIES = Arrays.asList(0.005, 0.01, 0.025, 0.05, 0.075, 0.1,
+            0.25, 0.5, 0.75, 1d, 2.5, 5d, 7.5, 10d);
     private final TelemetryMetricID metricId;
+    private final double MINIMUM_UPPER_BOUND_SECONDS;
 
     public TelemetryHistogramMetric(TelemetryMetricID metricId) {
         this.metricId = metricId;
+        MINIMUM_UPPER_BOUND_SECONDS = TCKConfig.getConfig().getTimeoutInMillis(200) / 1_000d;
     }
 
-    @SuppressWarnings("unchecked") // Fail early on a misconfigured test
-    public Collection<HistogramPointData> getHistogramPoints() {
+    public Optional<HistogramPointData> getHistogramPoint() {
         InMemoryMetricReader reader = InMemoryMetricReader.current();
-        return (List<HistogramPointData>) reader.getPointData(metricId);
+        return reader.getMetric(metricId)
+                .flatMap(md -> InMemoryMetricReader.getHistogramPointData(md, metricId));
     }
 
     public Optional<Long> getHistogramCount() {
-        Collection<HistogramPointData> hpd = getHistogramPoints();
+        return getHistogramPoint()
+                .map(HistogramPointData::getCount);
+    }
+
+    public boolean isPresent() {
+        return getHistogramPoint().isPresent();
+    }
+
+    public void assertBoundaries() {
+        HistogramPointData pointData =
+                getHistogramPoint().orElseThrow(() -> new AssertionError("No data point for " + metricId.name));
```

Comment:
Can this output metricId.toString() please.

Reply:
Done